lib/netext/dialer.go

```diff
@@ -173,6 +190,11 @@ func (d *Dialer) findRemote(addr string) (*lib.HostAddress, error) {
 		return lib.NewHostAddress(ip, port)
 	}
 
+	if d.BlockedHostnames != nil {
+		if match, blocked := d.BlockedHostnames.Contains(host); blocked {
+			return nil, BlockedHostError{hostname: host, match: match}
+		}
+	}
```

Comment:
Why is this block repeated here?

Reply:
I think because the other branch checks against `options.hosts`, though a comment wouldn't be amiss there...

Reply:
The idea was that I didn't want to parse the IP unless I needed to but looking at this code again ... the "speed up" in the case that there is no blockhostname and the host is configured in `options.host` will likely be miniscule and is the only case where there is a speed up so ... I am just going to do it before the options.host logic

Reply:
fixed in ce17b17a